tests/unit/conditions/test_return_value.py

```diff
@@ -1,14 +1,23 @@
+import os
 import pytest
 
 from nucypher.policy.conditions.lingo import ReturnValueTest
 
 
 def test_return_result_test_simple():
-    test = ReturnValueTest(comparator='>', value=0)
-    assert test.eval(1)
-    assert not test.eval(-1)
+    test = ReturnValueTest(comparator='>', value='0')
+    assert test.eval('1')
+    assert not test.eval('-1')
 
 
 def test_return_value_sanitization():
     with pytest.raises(ValueError):
         _test = ReturnValueTest('DROP', 'TABLE')
+
+def test_eval_is_evil():
+    file = f"file_{os.urandom(10).hex()}"
+    assert not os.path.isfile(file)
+    test = ReturnValueTest(comparator='>', value=f"(lambda: (42, __import__('os').system('touch {file}'))[0])()")
+    with pytest.raises(ValueError):
+        _ = test.eval('1000')
+    assert not os.path.isfile(file)
```

Comment:
It's good that this attack is no longer exploitable but we can take it a step further by raising an error as early as possible (in `ReturnValueTest.__init__`) if the value is not acceptable, in another PR of course.

Reply:
Yep, that's related to https://github.com/nucypher/nucypher/pull/2972#discussion_r977898875.